### ISSUE TITLE
Episodes not marked as watched

### ISSUE THREAD

**Issue by juanma-cvega** [jurialmunkey/plugin.video.themoviedb.helper#328]
I cannot get videos to get marked as watched, neither on library nor while browsing through the addon. If this is something that works for you I can try again and provide a log file.

**Comment by jurialmunkey** [comment]
Yeah it works fine for me.

Library strm files will mark as watched based on progress (i.e. watched more than 80% should mark watched) - but note that only the item in the library gets marked as watched and it only gets marked as watched when you play that particular strm file (e.g. if you play something from inside TMDbHelper it _won't_ mark your strm in the library as watched or vice-versa).

Inside the plugin the watched indicators are tied to your Trakt account. It's up to the particular plugin used to play the file to set the Trakt progress etc.

**Comment by Wafflestyx** [comment]
This might be related to 327.
I noticed when it would try to replay it would leave the library file as unwatched. 
Ending the video (before it fully ended) would mark it as watched (and not trigger the replay)

**Comment by juanma-cvega** [comment]
I'll test it again but I think it wasn't working as you describe it. I tested it watching an anime episode just until the credits and it didn't mark it as watched. And from the addon it's weird it doesn't do it because Venom used to do it fine. I'll try watching a video from Venom directly to see what it does.

**Comment by jurialmunkey** [comment]
@juanma-cvega is this still not working with latest version from testing branch?

**Comment by juanma-cvega** [comment]
Sorry, I haven't had the time to run tests lately. I think there is something weird but I need to run some more tests with the latest version you've published. I just watched an episode of an hour and it worked fine but yesterday I watched some anime episodes of 20 minutes and sometimes it worked and sometimes it didn't (specially with the setResolveURL, with that one I couldn't make it work once). I'm not sure if this is could be related to the remaining episode time when I stop the video. For anime there is always like a couple of minutes of credits I don't watch. But sometimes, even after leaving it running almost until the end it didn't work. 
I just got the new update with all your changes so once I get the TV for me to run more tests I'll let you now.
